pkg/loop/internal/pb/relayer.proto

```diff
@@ -0,0 +1,306 @@
+
+  rpc ChainStatus (ChainStatusRequest) returns (ChainStatusReply) {}
+  rpc ChainStatuses (ChainStatusesRequest) returns (ChainStatusesReply) {}
+
+  rpc NodeStatuses (NodeStatusesRequest) returns (NodeStatusesReply) {}
+
+  rpc SendTx (SendTxRequest) returns (google.protobuf.Empty) {}
+}
+
+// RelayArgs represents [github.com/smartcontractkit/chainlink-relay/pkg/types.RelayArgs].
+message RelayArgs {
+  bytes externalJobID = 1; // [32]byte
+  int32 jobID = 2;
+  string contractID = 3;
+  bool new = 4;
+  bytes relayConfig = 5;
+}
+
+// RelayArgs represents [github.com/smartcontractkit/chainlink-relay/pkg/types.PluginArgs].
+message PluginArgs {
+  string transmitterID = 1;
+  bytes pluginConfig = 2;
+}
+
+// NewConfigProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewConfigProvider].
+message NewConfigProviderRequest {
+  RelayArgs relayArgs = 1;
+}
+
+// NewConfigProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewConfigProvider].
+message NewConfigProviderReply {
+  uint32 configProviderID = 1;
+}
+
+// NewMedianProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMedianProvider].
+message NewMedianProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+// NewMedianProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMedianProvider].
+message NewMedianProviderReply {
+  uint32 medianProviderID = 1;
+}
+
+// NewMercuryProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMercuryProvider].
+message NewMercuryProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+// NewMercuryProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMercuryProvider].
+message NewMercuryProviderReply {
+  uint32 mercuryProviderID = 1;
+}
+
+// ChainStatusRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatus].
+message ChainStatusRequest {
+  string id = 1;
+}
+
+// ChainStatusReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatus].
+message ChainStatusReply {
+  ChainStatus chain = 1;
+}
+
+// ChainStatusesRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatuses].
+message ChainStatusesRequest {
+  int32 offset = 1;
+  int32 limit = 2;
+}
+
+// ChainStatusesReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatuses].
+message ChainStatusesReply {
+  repeated ChainStatus chains = 1;
+  int32 count = 2; // total count of chains
+}
+
+// ChainStatus represents [github.com/smartcontractkit/chainlink-relay/pkg/types.ChainStatus].
+message ChainStatus {
+  string id = 1;
+  bool enabled = 2;
+  string config = 3; // TOML
```

Comment:
So the whole TOML is represented as a string here?

Could we convert TOML into a protobuff?

Also, which configs will this include?
Chain-specific, and node specific? Or only 1?

Reply:
> So the whole TOML is represented as a string here?

Just for this chain.

> Could we convert TOML into a protobuff?

For what purpose? It would prevent relayers from owning the schema in the future, and it is not currently necessary since we are just displaying to the user.

> Also, which configs will this include?
> Chain-specific, and node specific? Or only 1?

I'm not sure what you mean. This isn't really changing from the old way much at all, which was a bag of JSON for a particular chain. Now we just have dynamic TOML instead of dynamic JSON.

Reply:
Ok, relayers owning the schema sounds good.

I don't know the config internals, but I was assuming some config values are not chain scoped, but global to the Node.
Was asking if those will be passed onto every plugin. But it is fine, if for now, this is mirroring the existing setup.

Reply:
The relayer internals do not currently depend on any global/root config fields directly, but we can accommodate that in the future if we need to. Although it is common to have it be overridable per-chain anyways, so we may not have to represent it separately.